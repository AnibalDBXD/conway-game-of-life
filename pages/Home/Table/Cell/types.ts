export interface ICellProps {
    isActive: boolean;
    onClick?: () => void;
}