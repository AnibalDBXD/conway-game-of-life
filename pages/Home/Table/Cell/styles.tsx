import styled from "styled-components";
import { COLORS } from "../../../../common/styles";
import { ICellProps } from "./types";

export const CellSize = 25;

export const StyledCell = styled.td<ICellProps>`
    background-color: ${(props): string => props.isActive ? COLORS.WHITE : COLORS.BLACK};
    min-width: ${CellSize | 25}px;
    height: ${CellSize | 25}px;
    cursor: pointer;
`;