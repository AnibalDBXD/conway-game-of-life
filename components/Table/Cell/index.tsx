import { StyledCell } from './styles';
import { ICellProps } from './types';

const Cell: React.FC<ICellProps> = ({
  isActive,
  onClick,
  'data-testid': testid,
}): JSX.Element => {
  return (
    <StyledCell onClick={onClick} data-testid={testid} isActive={isActive} />
  );
};

export default Cell;
