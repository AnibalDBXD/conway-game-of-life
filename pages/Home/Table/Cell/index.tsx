import { StyledCell } from "./styles";
import { ICellProps } from "./types";

const Cell: React.FC<ICellProps> = ({ isActive, onClick }): JSX.Element => {
  return (
    <StyledCell onClick={onClick} isActive={isActive} />
  );
};

export default Cell;
