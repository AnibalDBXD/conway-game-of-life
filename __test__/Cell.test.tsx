import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Cell from '../components/Table/Cell'
import 'jest-styled-components'

const Id = 'id'
const CellContainer = ({
  handleClick,
}: {
  handleClick?: () => void
}): JSX.Element => (
  <table>
    <tbody>
      <tr>
        <Cell isActive={false} onClick={handleClick} data-testid={Id} />
      </tr>
    </tbody>
  </table>
)

describe('<Cell />', () => {
  test('Render', () => {
    const { asFragment } = render(<CellContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('handle Click', () => {
    const mockHandler = jest.fn()
    const { getByTestId } = render(<CellContainer handleClick={mockHandler} />)
    const cell00 = getByTestId(Id)
    fireEvent.click(cell00)
    expect(mockHandler).toBeCalledTimes(1)
  })
})
