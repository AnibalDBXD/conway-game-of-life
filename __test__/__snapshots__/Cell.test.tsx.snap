// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Render 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #252525;
  min-width: 25px;
  height: 25px;
  cursor: pointer;
}

<table>
    <tbody>
      <tr>
        <td
          class="c0"
          data-testid="id"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
